### ISSUE TITLE
Framework: ATDM configuration should update the way Kokkos arch(es) are set

### ISSUE THREAD

**Issue by brian-kelley** [trilinos/Trilinos#7247]
## Bug Report
@trilinos/framework 

### Description
What went wrong?  What should have happened?  Do you have an idea what might
fix things?  How has this bug affected you?  What are you trying to accomplish?
What specifically must we demonstrate to call this work complete?

The ATDM environment scripts use the deprecated syntax ``-DKOKKOS_ARCH=XYZ``. After Kokkos 3.0, this should be ``-DKokkos_ARCH_XYZ=ON``. If there are multiple architectures enabled (like CPU and GPU) there should be multiple arch enable flags.

### Steps to Reproduce
Example on Stria:

```
source $TRILINOS_SRC/cmake/std/atdm/load-env.sh stria-arm-release
cmake DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=$TRILINOS_SRC/cmake/std/atdm/ATDMDevEnv.cmake ...... $TRILINOS_SRC

# This prints out warning:

CMake Warning at packages/kokkos/cmake/kokkos_functions.cmake:33 (MESSAGE):
  Given deprecated option list KOKKOS_ARCH.  This must now be given as
  separate -D options, which assuming you spelled options correctly would be:

    -DKokkos_ARCH_ARMV8-TX2=ON

  Remove CMakeCache.txt and re-run.  For a list of valid options, refer to
  BUILD.md or even look at CMakeCache.txt (before deleting it).
Call Stack (most recent call first):
  packages/kokkos/cmake/kokkos_arch.cmake:25 (KOKKOS_DEPRECATED_LIST)
  packages/kokkos/cmake/kokkos_tribits.cmake:181 (INCLUDE)
  packages/kokkos/CMakeLists.txt:156 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
```

**Comment by prwolfe** [comment]
Actually these are maintained by @bartlettroscoe

**Comment by bartlettroscoe** [comment]
@brian-kelley, yea, I noticed that.  That should not be too hard to fix.  I have put this on the "Selected" list under https://github.com/trilinos/Trilinos/projects/6.

**Comment by bartlettroscoe** [comment]
With PR #7448, this should be done.  (Thanks @jjellio!)

Putting in review.

**Comment by bartlettroscoe** [comment]
This was addressed in PR #7448 and has not seemed to be causing any problems.

Closing as complete.
